plugins/kubernetes/app/controllers/kubernetes_dashboard_controller.rb

```diff
@@ -0,0 +1,82 @@
+require 'kubeclient'
+
+class KubernetesDashboardController < ApplicationController
+  include CurrentProject
+
+  def index
+    render json: format_pod_info(pod_info).to_json
+  end
+
+  private
+
+  def pod_info
+    roles = {}
+    Environment.find(params[:environment]).cluster_deploy_groups.each do |cluster_deploy_group|
+      cluster_deploy_group.cluster.client.get_pods(
+          namespace: cluster_deploy_group.namespace,
+          label_selector: "project=#{current_project.name_for_label}").each do |pod|
+        labels = pod.metadata.labels
+        roles[labels.role_id] ||= build_role(labels.role_id)
+        role = roles[labels.role_id]
+        deploy_group_id = cluster_deploy_group.deploy_group.id
+        role[:deploy_groups][deploy_group_id] ||= build_deploy_group(deploy_group_id)
+        deploy_group = role[:deploy_groups][deploy_group_id]
+        deploy_group[:releases][labels.release_id] ||= build_release(labels.release_id, labels.role_id)
+        release = deploy_group[:releases][labels.release_id]
+        api_pod = Kubernetes::Api::Pod.new(pod)
+        release[:live_replicas] += 1 if api_pod.ready?
+      end
+    end
+    roles
+  end
+
+  def build_role(role_id)
+    {
+        name: role_name(role_id),
+        deploy_groups: {}
+    }
+  end
+
+  def build_deploy_group(deploy_group_id)
+    {
+        name: deploy_group_name(deploy_group_id),
+        releases: {}
+    }
+  end
+
+  def build_release(release_id, role_id)
+    {
+        id: release_id,
+        build: build_label(release_id),
+        target_replicas: target_replicas(release_id, role_id),
+        live_replicas: 0
+    }
+  end
+
+  def role_name(role_id)
+    Kubernetes::Role.find(role_id).name
```

Comment:
Isn't the role name one of the metadata labels we're sending to Kubernetes? Not sure, though...